src/display/canvas.js

```diff
@@ -871,6 +874,21 @@ function composeSMask(ctx, smask, layerCtx) {
   ctx.drawImage(mask, 0, 0);
 }
 
+function getImageSmoothingEnabled(transform, interpolate) {
+  const scale = Util.singularValueDecompose2dScale(transform);
+  const actualScale =
+    (window.devicePixelRatio * CSS_PIXELS_PER_INCH) / PDF_PIXELS_PER_INCH;
```

Comment:
This is causing the unit-tests to fail, when run in Node.js, since we cannot assume that `window` is defined.
How about simply doing `globalThis.devicePixelRatio || 1` here, since that ought to prevent any issues?